pkg/securelaunch/helpers.go

```diff
@@ -0,0 +1,270 @@
+	}
+	Debug("WriteToFile exit w success data written to target=%s", target)
+	return target, nil
+}
+
+/*
+ * src and dst are already mounted file paths.
+ * src and dst are Absolute file paths
+ * defFileName is default dst file name, only used if user doesn't provide one.
+ * returns
+ * 1. target file path where file was written
+ * 2. error
+ */
+func WriteSrcFiletoDst(src, dst, defFileName string) (string, error) {
+	srcFileFound, _, err := exists(src)
+	if !srcFileFound {
+		return "", fmt.Errorf("File to be written doesnt exist:")
+	}
+
+	d, err := ioutil.ReadFile(src)
+	if err != nil {
+		return "", fmt.Errorf("Error reading src file %s, err=%v", src, err)
+	}
+
+	target, err := WriteToFile(d, dst, defFileName)
+	if err != nil {
+		return "", fmt.Errorf("WriteSrcFiletoDst failed err=%v", err)
+	}
+
+	Debug("WriteSrcFileToDst exit w success src=%s written to target=%s", src, target)
+	return target, nil
+}
+
+/*
+ * NOTE: Caller's responsbility to unmount this..use return var mountPath to unmount in caller.
+ *
+ * obtains absolute file path based on input entered by user in policy file.
+ * inputVal is of format <block device identifier>:<path>
+ * inputVal e.g sda:/boot/foo.go , 4qccd342-12zr-4e99-9ze7-1234cb1234c4:/bar/zyx.go
+ * rw_option = true --> mount is of type read-write, else read-only.
+ *
+ * This function
+ * parses user input.
+ * mounts device
+ * Get absolute file path on mounted device.
+ * does NOT Unmount device, use returned mountPath to unmount...
+ * returns filePath, mountPath, error in that order
+ */
+func GetMountedFilePath(inputVal string, rw_option bool) (string, string, error) {
+	s := strings.Split(inputVal, ":")
+	if len(s) != 2 {
+		return "", "", fmt.Errorf("%s: Usage: <block device identifier>:<path>", inputVal)
+	}
+
+	// s[0] can be sda or UUID. if UUID, then we need to find its name
+	var deviceId string = s[0]
+	if !strings.HasPrefix(deviceId, "sd") {
+		if e := getBlkInfo(); e != nil {
+			return "", "", fmt.Errorf("GetMountedFilePath: getBlkInfo err=%s", e)
+		}
+		devices := storage.PartitionsByFsUUID(storageBlkDevices, s[0]) // []BlockDev
+		for _, device := range devices {
+			Debug("device =%s with fsuuid=%s", device.Name, s[0])
+			deviceId = device.Name
+		}
+	}
+
+	devicePath := filepath.Join("/dev", deviceId) // assumes deviceId is sda, devicePath=/dev/sda
+	Debug("Attempting to mount %s", devicePath)
+	var dev *diskboot.Device
+	var err error
+	if rw_option {
+		dev, err = diskboot.FindDeviceRW(devicePath) // FindDevice fn mounts , w rw option, devicePath=/dev/sda.
+	} else {
+		dev, err = diskboot.FindDevice(devicePath) // FindDevice fn mounts devicePath=/dev/sda.
+	}
+	if err != nil {
+		return "", "", fmt.Errorf("Mount %v failed, err=%v", devicePath, err)
+	}
+
+	Debug("Mounted %s", devicePath)
+	fPath := filepath.Join(dev.MountPath, s[1]) // mountPath=/tmp/path/to/target/file if /dev/sda mounted on /tmp
+	return fPath, dev.MountPath, nil
+}
+
+/*
+ * ScanKernelCmdLine() ([]byte, error)
```

Comment:
same, re comment. I'd also perhaps rename this? The name doesn't say what you're scanning the cmdline for

```go
// LocatePolicy locates the policy file via the kernel cmdline.
//
// Cmdline format: sl_policy=<...
func LocatePolicy([]byte, error) {
```

Reply:
LocatePolicy
1. checks for policy file in kernel cmdline performed by ScanKernelCmdLine if its found there, it returns with success.
2. if its not found, searches for policy file in all block devices and for each block device, searches for policy file under "/", "/efi" or "/boot", this is performed by ScanBlockDevice,

Reply:
I can move the ScanKernelCmdLine and ScanBlockDevice functions over to the policy sub-package if that makes it easier for reading ? They all call the functions in the helpers.go file , so i was trying to avoid seeing code like slaunch.HelperFunction all around...

Reply:
Moving it sounds good to me. Maybe from there, you can make them unexported internal functions? `scanKernelCmdline` and `scanBlockDevice`

Reply:
makes sense